### PR TITLE
Volatile & Reloadable.distinctByKey

### PR DESCRIPTION
**Pull request by danilbykov**
1. Bump cats version
2. Add new method `Reloadable.distinctByKey` to filter out duplicated events
3. Add new trait `Volatile` which is simplified `Reloadable` but `Volatile` has method `mapK` which allows to change its context. This method can not be implemented on `Reloadable` because effect `F` is contained in both covariant and contravariant positions. `Volatile` has simpler interface with `F` only in covariant position.

### CODE REVIEW

File: core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/ReloadBehaviour.scala [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+import cats.~>
+
 /**
   * Reload policy
   *
   * @see [[ReloadableInternal]]
   */
-sealed trait ReloadBehaviour[F[_], A, B]
+sealed trait ReloadBehaviour[F[_], A, B] {
+  def mapK[G[_]](fk: F ~> G): ReloadBehaviour[G, A, B]
```
Comment (fit51) [inline core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/ReloadBehaviour.scala]: In the end, this method is not used.

**Review by fit51 — CHANGES_REQUESTED** [review]
Please, remove mapK from ReloadBehaviour

File: build.sbt [hunk 2]
```diff
@@ -88,7 +91,7 @@ lazy val examples = project
     libraryDependencies ++= Seq(
       "ch.qos.logback" % "logback-classic" % "1.2.3",
       "ch.qos.logback" % "logback-core"    % "1.2.3",
-      "io.circe"       %% "circe-generic"  % "0.12.2"
+      "io.circe"       %% "circe-generic"  % circeVersion % Provided
```
Comment (fit51) [inline build.sbt:94]:
```suggestion
      "io.circe"       %% "circe-generic"  % circeVersion,
      "io.circe"       %% "circe-parser"   % circeVersion
```

File: core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/ReloadBehaviour.scala (hunk 1 above)
Comment (danilbykov) [reply core/src/main/scala/com/github/fit51/reactiveconfig/reloadable/ReloadBehaviour.scala]: Done.

**Review by fit51 — APPROVED** [review]

**Pull request merged**